src/components/tiles/table/table-tile.tsx

```diff
@@ -69,6 +71,25 @@ const TableToolComponent: React.FC<ITileProps> = observer(function TableToolComp
     ref: gridRef, gridContext, inputRowId, selectedCell, getSelectedRows, ...gridProps
   } = useGridContext({ modelId: model.id, showRowLabels, triggerColumnChange });
 
+  // Maintains the cache of data values that map to image URLs.
+  // For use in a synchronous context, returns undefined immediately if an image is not yet cached,
+  // and then looks it up in the background, adds to cache, and updates state to force a refresh.
+  const lookupImage = (value: string) => {
+    if (gImageMap.isImageUrl(value)) {
+      const cached = imageUrls.get(value);
+      if (cached) {
+        return cached;
+      }
+      gImageMap.getImage(value).then((image) => {
+        if (image && image.displayUrl) {
+          // This state changes forces a re-render - is that good?
+          setImageUrls(new Map(imageUrls).set(value, image.displayUrl));
```

Comment:
I agree that this feels a little weird. If you need to force a rerender here (which seems likely), I think there are a couple of functions in the table tile like `updateCases` and `updateAttributes` (probably not exactly that but something similar) which just increment a counter somewhere. That might make more sense than duplicating the map every time it gets a new item. That said, this is probably not going to cause performance issues as is so I wouldn't worry too much about it.

Reply:
I'll leave this as is then - it is semantically reasonable to consider the map to be state.